// yaml/browser/dist/compose/resolve-flow-scalar.js
neric Events\n  topPress: {\n    phasedRegistrationNames: {\n      bubbled: 'onPress',\n      captured: 'onPressCapture',\n    },\n  },\n  topChange: {\n    phasedRegistrationNames: {\n      bubbled: 'onChange',\n      captured: 'onChangeCapture',\n    },\n  },\n  topFocus: {\n    phasedRegistrationNames: {\n      bubbled: 'onFocus',\n      captured: 'onFocusCapture',\n    },\n  },\n  topBlur: {\n    phasedRegistrationNames: {\n      bubbled: 'onBlur',\n      captured: 'onBlurCapture',\n    },\n  },\n  topSubmitEditing: {\n    phasedRegistrationNames: {\n      bubbled: 'onSubmitEditing',\n      captured: 'onSubmitEditingCapture',\n    },\n  },\n  topEndEditing: {\n    phasedRegistrationNames: {\n      bubbled: 'onEndEditing',\n      captured: 'onEndEditingCapture',\n    },\n  },\n  topKeyPress: {\n    phasedRegistrationNames: {\n      bubbled: 'onKeyPress',\n      captured: 'onKeyPressCapture',\n    },\n  },\n\n  // Touch Events\n  topTouchStart: {\n    phasedRegistrationNames: {\n      bubbled: 'onTouchStart',\n      captured: 'onTouchStartCapture',\n    },\n  },\n  topTouchMove: {\n    phasedRegistrationNames: {\n      bubbled: 'onTouchMove',\n      captured: 'onTouchMoveCapture',\n    },\n  },\n  topTouchCancel: {\n    phasedRegistrationNames: {\n      bubbled: 'onTouchCancel',\n      captured: 'onTouchCancelCapture',\n    },\n  },\n  topTouchEnd: {\n    phasedRegistrationNames: {\n      bubbled: 'onTouchEnd',\n      captured: 'onTouchEndCapture',\n    },\n  },\n\n  // Experimental/Work in Progress Pointer Events (not yet ready for use)\n  topPointerCancel: {\n    phasedRegistrationNames: {\n      captured: 'onPointerCancelCapture',\n      bubbled: 'onPointerCancel',\n    },\n  },\n  topPointerDown: {\n    phasedRegistrationNames: {\n      captured: 'onPointerDownCapture',\n      bubbled: 'onPointerDown',\n    },\n  },\n  topPointerMove: {\n    phasedRegistrationNames: {\n      captured: 'onPointerMoveCapture',\n      bubbled: 'onPointerMove',\n    },\n  },\n  topPointerUp: {\n    phasedRegistrationNames: {\n      captured: 'onPointerUpCapture',\n      bubbled: 'onPointerUp',\n    },\n  },\n  topPointerEnter: {\n    phasedRegistrationNames: {\n      captured: 'onPointerEnterCapture',\n      bubbled: 'onPointerEnter',\n      skipBubbling: true,\n    },\n  },\n  topPointerLeave: {\n    phasedRegistrationNames: {\n      captured: 'onPointerLeaveCapture',\n      bubbled: 'onPointerLeave',\n      skipBubbling: true,\n    },\n  },\n  topPointerOver: {\n    phasedRegistrationNames: {\n      captured: 'onPointerOverCapture',\n      bubbled: 'onPointerOver',\n    },\n  },\n  topPointerOut: {\n    phasedRegistrationNames: {\n      captured: 'onPointerOutCapture',\n      bubbled: 'onPointerOut',\n    },\n  },\n};\n\nconst directEventTypes = {\n  topAccessibilityAction: {\n    registrationName: 'onAccessibilityAction',\n  },\n  topAccessibilityTap: {\n    registrationName: 'onAccessibilityTap',\n  },\n  topMagicTap: {\n    registrationName: 'onMagicTap',\n  },\n  topAccessibilityEscape: {\n    registrationName: 'onAccessibilityEscape',\n  },\n  topLayout: {\n    registrationName: 'onLayout',\n  },\n  onGestureHandlerEvent: DynamicallyInjectedByGestureHandler({\n    registrationName: 'onGestureHandlerEvent',\n  }),\n  onGestureHandlerStateChange: DynamicallyInjectedByGestureHandler({\n    registrationName: 'onGestureHandlerStateChange',\n  }),\n  // [Windows\n  // Mouse enter/leave events\n  topMouseEnter: {\n    registrationName: 'onMouseEnter',\n  },\n  topMouseLeave: {\n    registrationName: 'onMouseLeave',\n  },\n  // Windows]\n};\n\nconst validAttributesForNonEventProps = {\n  // View Props\n  accessible: true,\n  accessibilityActions: true,\n  accessibilityLabel: true,\n  accessibilityHint: true,\n  accessibilityLanguage: true,\n  accessibilityValue: true,\n  accessibilityViewIsModal: true,\n  accessibilityElementsHidden: true,\n  accessibilityIgnoresInvertColors: true,\n  testID: true,\n  backgroundColor: {process: require('../StyleSheet/processColor')},\n  backfaceVisibility: true,\n  opacity: true,\n  shadowColor: {proces)
                res += cc;
            else if (next === '\n') {
                // skip escaped newlines, but still trim the following line
                next = source[i + 1];
                while (next === ' ' || next === '\t')
                    next = source[++i + 1];
            }
            else if (next === '\r' && source[i + 1] === '\n') {
                // skip escaped CRLF newlines, but still trim the following line
                next = source[++i + 1];
                while (next === ' ' || next === '\t')
                    next = source[++i + 1];
            }
            else if (next === 'x' || next === 'u' || next === 'U') {
                const length = { x: 2, u: 4, U: 8 }[next];
                res += parseCharCode(source, i + 1, length, onError);
                i += length;
            }
            else {
                const raw = source.substr(i - 1, 2);
                onError(i - 1, 'BAD_DQ_ESCAPE', `Invalid escape sequence ${raw}`);
                res += raw;
            }
        }
        else if (ch === ' ' || ch === '\t') {
            // trim trailing whitespace
            const wsStart = i;
            let next = source[i + 1];
            while (next === ' ' || next === '\t')
                next = source[++i + 1];
            if (next !== '\n' && !(next === '\r' && source[i + 2] === '\n'))
                res += i > wsStart ? source.slice(wsStart, i + 1) : ch;
        }
        else {
            res += ch;
        }
    }
    if (source[source.length - 1] !== '"' || source.length === 1)
        onError(source.length, 'MISSING_CHAR', 'Missing closing "quote');
    return res;
}
/**
 * Fold a single newline into a space, multiple newlines to N - 1 newlines.
 * Presumes `source[offset] === '\n'`
 */
function foldNewline(source, offset) {
    let fold = '';
    let ch = source[offset + 1];
    while (ch === ' ' || ch === '\t' || ch === '\n' || ch === '\r') {
        if (ch === '\r' && source[offset + 2] !== '\n')
            break;
        if (ch === '\n')
            fold += '\n';
        offset += 1;
        ch = source[offset + 1];
    }
    if (!fold)
        fold = ' ';
    return { fold, offset };
}
const escapeCodes = {
    '0': '\0', // null character
    a: '\x07', // bell character
    b: '\b', // backspace
    e: '\x1b', // escape character
    f: '\f', // form feed
    n: '\n', // line feed
    r: '\r', // carriage return
    t: '\t', // horizontal tab
    v: '\v', // vertical tab
    N: '\u0085', // Unicode next line
    _: '\u00a0', // Unicode non-breaking space
    L: '\u2028', // Unicode line separator
    P: '\u2029', // Unicode paragraph separator
    ' ': ' ',
    '"': '"',
    '/': '/',
    '\\': '\\',
    '\t': '\t'
};
function parseCharCode(source, offset, length, onError) {
    const cc = source.substr(offset, length);
    const ok = cc.length === length && /^[0-9a-fA-F]+$/.test(cc);
    const code = ok ? parseInt(cc, 16) : NaN;
    if (isNaN(code)) {
        const raw = source.substr(offset - 2, length + 2);
        onError(offset - 2, 'BAD_DQ_ESCAPE', `Invalid escape sequence ${raw}`);
        return raw;
    }
    return String.fromCodePoint(code);
}

export { resolveFlowScalar };
